Compile Failures with GCC 15.1

I've noticed that when upgraded to GCC 15.1 from GCC 14.2, I see the following errors when running `make`:

```
+ make
make  all-recursive
Making all in workloads
Making all in src
...
mv -f .deps/protocol.Tpo .deps/protocol.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -Wno-error -O2 -D_REENTRANT -MT tcp.o -MD -MP -MF .deps/tcp.Tpo -c -o tcp.o tcp.c
mv -f .deps/tcp.Tpo .deps/tcp.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -Wno-error -O2 -D_REENTRANT -MT generic.o -MD -MP -MF .deps/generic.Tpo -c -o generic.o generic.c
mv -f .deps/generic.Tpo .deps/generic.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -Wno-error -O2 -D_REENTRANT -MT hwcounter.o -MD -MP -MF .deps/hwcounter.Tpo -c -o hwcounter.o hwcounter.c
hwcounter.c:82:1: error: conflicting types for 'hwcounter_fini'; have 'int(hwcounter_t *)'
   82 | hwcounter_fini(hwcounter_t * hw)
      | ^~~~~~~~~~~~~~
In file included from hwcounter.c:28:
hwcounter.h:63:5: note: previous declaration of 'hwcounter_fini' with type 'int(void)'
   63 | int hwcounter_fini();
      |     ^~~~~~~~~~~~~~
*** Error code 1
make: Fatal error: Command failed for target `hwcounter.o'
```

The header hwcounter.h defines hwcounter_fini like so:

`int hwcounter_fini()`

But the definition in hwcounter.c is:

`int hwcounter_fini(hwcounter_t *hw);`

Perhaps this is because according to the [GCC 15 changelog:](https://gcc.gnu.org/gcc-15/changes.html)

> GCC 15 changes the default language version for C compilation from [-std=gnu17](https://gcc.gnu.org/onlinedocs/gcc-15.1.0/gcc/C-Dialect-Options.html#index-std-1) to [-std=gnu23](https://gcc.gnu.org/onlinedocs/gcc-15.1.0/gcc/C-Dialect-Options.html#index-std-1).

## Workaround:

The following workaround allowed me to compile:

$ ./configure
$ gsed -i 's/int hwcounter_fini();/int hwcounter_fini(hwcounter_t *);/' src/hwcounter.h
$ make
```

make  all-recursive
Making all in workloads
Making all in src
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT workorder.o -MD -MP -MF .deps/workorder.Tpo -c -o workorder.o workorder.c
mv -f .deps/workorder.Tpo .deps/workorder.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT strand.o -MD -MP -MF .deps/strand.Tpo -c -o strand.o strand.c
mv -f .deps/strand.Tpo .deps/strand.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT execute.o -MD -MP -MF .deps/execute.Tpo -c -o execute.o execute.c
mv -f .deps/execute.Tpo .deps/execute.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT flowops_library.o -MD -MP -MF .deps/flowops_library.Tpo -c -o flowops_library.o flowops_library.c
mv -f .deps/flowops_library.Tpo .deps/flowops_library.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT flowops.o -MD -MP -MF .deps/flowops.Tpo -c -o flowops.o flowops.c
mv -f .deps/flowops.Tpo .deps/flowops.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT common.o -MD -MP -MF .deps/common.Tpo -c -o common.o common.c
mv -f .deps/common.Tpo .deps/common.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT main.o -MD -MP -MF .deps/main.Tpo -c -o main.o main.c
mv -f .deps/main.Tpo .deps/main.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT slave.o -MD -MP -MF .deps/slave.Tpo -c -o slave.o slave.c
mv -f .deps/slave.Tpo .deps/slave.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT stats.o -MD -MP -MF .deps/stats.Tpo -c -o stats.o stats.c
mv -f .deps/stats.Tpo .deps/stats.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT handshake.o -MD -MP -MF .deps/handshake.Tpo -c -o handshake.o handshake.c
mv -f .deps/handshake.Tpo .deps/handshake.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT parse.o -MD -MP -MF .deps/parse.Tpo -c -o parse.o parse.c
mv -f .deps/parse.Tpo .deps/parse.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT shm.o -MD -MP -MF .deps/shm.Tpo -c -o shm.o shm.c
mv -f .deps/shm.Tpo .deps/shm.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT master.o -MD -MP -MF .deps/master.Tpo -c -o master.o master.c
mv -f .deps/master.Tpo .deps/master.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT print.o -MD -MP -MF .deps/print.Tpo -c -o print.o print.c
mv -f .deps/print.Tpo .deps/print.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT signals.o -MD -MP -MF .deps/signals.Tpo -c -o signals.o signals.c
mv -f .deps/signals.Tpo .deps/signals.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT goodbye.o -MD -MP -MF .deps/goodbye.Tpo -c -o goodbye.o goodbye.c
mv -f .deps/goodbye.Tpo .deps/goodbye.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT delay.o -MD -MP -MF .deps/delay.Tpo -c -o delay.o delay.c
mv -f .deps/delay.Tpo .deps/delay.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT rate.o -MD -MP -MF .deps/rate.Tpo -c -o rate.o rate.c
mv -f .deps/rate.Tpo .deps/rate.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT sendfilev.o -MD -MP -MF .deps/sendfilev.Tpo -c -o sendfilev.o sendfilev.c
mv -f .deps/sendfilev.Tpo .deps/sendfilev.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT logging.o -MD -MP -MF .deps/logging.Tpo -c -o logging.o logging.c
mv -f .deps/logging.Tpo .deps/logging.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT netstat.o -MD -MP -MF .deps/netstat.Tpo -c -o netstat.o netstat.c
mv -f .deps/netstat.Tpo .deps/netstat.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT numbers.o -MD -MP -MF .deps/numbers.Tpo -c -o numbers.o numbers.c
mv -f .deps/numbers.Tpo .deps/numbers.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT sync.o -MD -MP -MF .deps/sync.Tpo -c -o sync.o sync.c
mv -f .deps/sync.Tpo .deps/sync.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT protocol.o -MD -MP -MF .deps/protocol.Tpo -c -o protocol.o protocol.c
mv -f .deps/protocol.Tpo .deps/protocol.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT tcp.o -MD -MP -MF .deps/tcp.Tpo -c -o tcp.o tcp.c
mv -f .deps/tcp.Tpo .deps/tcp.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT generic.o -MD -MP -MF .deps/generic.Tpo -c -o generic.o generic.c
mv -f .deps/generic.Tpo .deps/generic.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT hwcounter.o -MD -MP -MF .deps/hwcounter.Tpo -c -o hwcounter.o hwcounter.c
mv -f .deps/hwcounter.Tpo .deps/hwcounter.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT sctp.o -MD -MP -MF .deps/sctp.Tpo -c -o sctp.o sctp.c
mv -f .deps/sctp.Tpo .deps/sctp.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT ssl.o -MD -MP -MF .deps/ssl.Tpo -c -o ssl.o ssl.c
ssl.c: In function 'load_engine':
ssl.c:443:9: warning: 'ENGINE_load_builtin_engines' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  443 |         ENGINE_load_builtin_engines();
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ssl.c:39:
/usr/include/openssl/engine.h:358:28: note: declared here
  358 | OSSL_DEPRECATEDIN_3_0 void ENGINE_load_builtin_engines(void);
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
ssl.c:444:9: warning: 'ENGINE_by_id' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  444 |         e = ENGINE_by_id(engine_id);
      |         ^
/usr/include/openssl/engine.h:336:31: note: declared here
  336 | OSSL_DEPRECATEDIN_3_0 ENGINE *ENGINE_by_id(const char *id);
      |                               ^~~~~~~~~~~~
ssl.c:448:9: warning: 'ENGINE_register_complete' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  448 |         ENGINE_register_complete(e);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/openssl/engine.h:414:27: note: declared here
  414 | OSSL_DEPRECATEDIN_3_0 int ENGINE_register_complete(ENGINE *e);
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~
mv -f .deps/ssl.Tpo .deps/ssl.Po
gcc -DHAVE_CONFIG_H -I. -I..      -O2 -D_REENTRANT -MT udp.o -MD -MP -MF .deps/udp.Tpo -c -o udp.o udp.c
mv -f .deps/udp.Tpo .deps/udp.Po
gcc  -O2 -D_REENTRANT   -o uperf workorder.o strand.o  execute.o flowops_library.o flowops.o  common.o main.o slave.o  stats.o handshake.o parse.o  shm.o master.o print.o  signals.o goodbye.o delay.o  rate.o sendfilev.o logging.o  netstat.o numbers.o sync.o  protocol.o tcp.o generic.o  hwcounter.o sctp.o ssl.o  udp.o   -lrt -lm -lsocket -lnsl -lumem -lkstat -lsendfile -lcpc -lintl -lkstat -lsctp -lsctp -lcrypto -lssl -lssl  -lrt -lm -lsocket -lnsl -lumem -lkstat -lsendfile -lcpc -lintl -lkstat
Making all in tests
+ ret=0
```

--- a/src/hwcounter.h
+++ b/src/hwcounter.h
@@ -59,8 +59,8 @@ typedef struct {
 	int		init_status;
 }hwcounter_t;
 
-int hwcounter_init();
-int hwcounter_fini();
+int hwcounter_init(void);
+int hwcounter_fini(hwcounter_t *));
 int hwcounter_validate_events(char *, char *);
 int hwcounter_finilwp(hwcounter_t *);
 int hwcounter_snap(hwcounter_t *, int);

